### PR TITLE
refactor who_added_track

### PR DESCRIPTION
**Pull request by jbodah**

### CODE REVIEW

File: lib/spotify.rb [hunk 1]
```diff
@@ -102,6 +105,11 @@ def client_secret
       @config[:client_secret]
     end
 
+    # TODO
+    def default_playlist_uri
```
Comment (jbodah) [inline lib/spotify.rb:109]: I think we want to just rename `default_uri` to `default_playlist_uri`

File: lib/spotify/web.rb [hunk 2]
```diff
@@ -10,6 +10,7 @@ def search(type, q)
       cast RSpotify.const_get(type.capitalize).search(q)
     end
 
+    # TODO what type is artist? URI? ID? Name?
```
Comment (jbodah) [inline lib/spotify/web.rb:13]: e.g. docstrings

File: lib/spotify/web.rb [hunk 3]
```diff
@@ -18,27 +19,54 @@ def tracks_on_album(album)
...
       end
 
-      { name: all_songs[track_id].id }
+      # TODO - make a type for this
+      { name: added_by.id }
     end
 
     private
 
+    # TODO - what does this yield
```
Comment (jbodah) [inline lib/spotify/web.rb:50]: RSpotify::User

File: lib/spotify/web.rb [hunk 4]
```diff
@@ -18,27 +19,54 @@ def tracks_on_album(album)
...
+    def playlist_tracks_by_added_by(playlist_uri)
+      Enumerator.new do |yielder|
```
Comment (jbodah) [inline lib/spotify/web.rb:52]: TODO: does Enumerator fully enumerate or is it lazy?

File: lib/spotify/web.rb [hunk 5]
```diff
@@ -18,27 +19,54 @@ def tracks_on_album(album)
...
+        # TODO - extract URI type with helpers for parsing
+        user, playlist_id = playlist_uri.split(':').each_slice(2).map(&:last)
+        playlist = RSpotify::Playlist.find(user, playlist_id)
+        offset = 0
+        # Fetch batch of tracks
+        tracks = playlist.tracks(offset: offset)
+        until tracks.empty? do
+          # TODO - what is this??
```
Comment (jbodah) [inline lib/spotify/web.rb:60]: track_id to RSpotify::User hash

File: lib/spotify/web.rb [hunk 6]
```diff
@@ -18,27 +19,54 @@ def tracks_on_album(album)
...
+    #end
+
+    def who_added_track(playlist_uri, track_id)
+      added_by = playlist_tracks_by_added_by(playlist_uri).find do |t_id, added_by|
+        t_id == track_id
       end
 
-      { name: all_songs[track_id].id }
+      # TODO - make a type for this
+      { name: added_by.id }
```
Comment (jbodah) [inline lib/spotify/web.rb:45]: https://github.com/guilhermesad/rspotify/blob/master/lib/rspotify/user.rb#L6

File: lib/spotify.rb (hunk 1 above)
Comment (thebmo) [inline lib/spotify.rb:109]: would be more clear :+1:

**Pull request closed without merging**